### PR TITLE
Add benchmark folder for PR testing

### PR DESCRIPTION
**Pull request by JSOBot**

### CODE REVIEW

**Comment by MonssafToukal** [comment]
@dpo I created a new PR specifically for the benchmark files.

File: benchmark/run_benchmarks.jl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+# extract stats for each benchmark to plot profiles
+# files_dict will be part of json_dict below
+files_dict = Dict{String, Any}()
+file_num = 1
+for k ∈ keys(judgement_stats)
+  global file_num
+  k_stats = Dict{Symbol,DataFrame}(:commit => commit_stats[k],
+                                   :master => master_stats[k])
+  save_stats(k_stats, "ldl_$(bmarkname)_vs_master_$(k).jld2", force=true)
```
Comment (tmigot) [inline benchmark/run_benchmarks.jl:51]: Why is it `ldl` here?

File: benchmark/run_benchmarks.jl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  file_num += 1
+end
+
+for mdfile ∈ [:judgement, :master, :commit]
+  global file_num
+  files_dict["$(file_num)_$(mdfile).md"] = Dict{String,Any}("content" => "$(sprint(export_markdown, eval(mdfile)))")
+  file_num += 1
+end
+
+jldopen("ldl_$(bmarkname)_vs_master_judgement.jld2", "w") do file
```
Comment (tmigot) [inline benchmark/run_benchmarks.jl:70]: see question above.

File: benchmark/send_comment_to_pr.jl [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+DEFAULT_GIST_FILE_PATH = "./gist.json"
+
+function parse_commandline()
+    s = ArgParseSettings()
+    @add_arg_table! s begin
+        "--org", "-o"
+            help = "Name of GitHub Organization"
+            arg_type = String
+            default = "ProofOfConceptForJuliSmoothOptimizers"
```
Comment (tmigot) [inline benchmark/send_comment_to_pr.jl:18]: Should this be `JuliaSmoothOptimizers?

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            expressionType: 'JSONPath', //Optional, defaults to JSONPath
+            regexpFilter: '', //Optional, defaults to empty string
+            defaultValue: '' //Optional, defaults to empty string
+        ],
+        [
+            key: 'org',
+            value: '$.organization.login',
+            expressionType: 'JSONPath', //Optional, defaults to JSONPath
+            regexpFilter: '', //Optional, defaults to empty string
+            defaultValue: 'ProofOfConceptForJuliSmoothOptimizers' //Optional, defaults to empty string
```
Comment (tmigot) [inline Jenkinsfile:32]: I think we concluded in #23 that 'ProofOfConceptForJuliSmoothOptimizers' should be replaced by 'JuliaSmoothOptimizers'

File: benchmark/run_benchmarks.jl (hunk 1 above)
Comment (MonssafToukal) [reply benchmark/run_benchmarks.jl:51]: There shouldn't be an `ldl` here. I will make the necessary modifications on my repository. Thank you!

File: Jenkinsfile (hunk 4 above)
Comment (MonssafToukal) [reply Jenkinsfile:32]: Yes, exactly! I will change it on my side as well.

File: benchmark/run_benchmarks.jl (hunk 1 above)
Comment (MonssafToukal) [reply benchmark/run_benchmarks.jl:51]: `$(bmarkname)_vs_master_$(k).jld2` should be enough.

**Pull request closed without merging**